specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -162,131 +166,131 @@ namespace Azure.Messaging.EventGrid {
 
-  @doc("Acknowledge batch of Cloud Events. The server responds with an HTTP 200 status code if at least one event is successfully acknowledged. The response body will include the set of successfully acknowledged lockTokens, along with other failed lockTokens with their corresponding error information. Successfully acknowledged events will no longer be available to any consumer.")
-
-  @route("/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:acknowledge")
-  @post op AcknowledgeCloudEvents is Azure.Core.RpcOperation<{
-    @doc("content type")
-    @header("content-type")
-    contentType: "application/json; charset=utf-8";
-
-    @doc("Topic Name.")
-    @path
-    topicName: string;
 
-    @doc("Event Subscription Name.")
-    @path
-    eventSubscriptionName: string;
-
-    @doc("AcknowledgeOptions.")
-    @body
-    lockTokens: AcknowledgeOptions;
-  }, AcknowledgeResult>;
+  @doc("Acknowledge batch of Cloud Events. The server responds with an HTTP 200 status code if at least one event is successfully acknowledged. The response body will include the set of successfully acknowledged lockTokens, along with other failed lockTokens with their corresponding error information. Successfully acknowledged events will no longer be available to any consumer.")
+  @action("acknowledge")
+  op acknowledgeCloudEvents is StandardOperations.ResourceAction<
+    EventSubscription,
+    {
+      @doc("content type")
+      @header("content-type")
+      contentType: "application/json; charset=utf-8";   
+
+      @doc("AcknowledgeOptions.")
+      @body
+      lockTokens: AcknowledgeOptions;
+    },
+    AcknowledgeResult
+  >;
 
   // Release Operation:
   // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:release?api-version={apiVersion}
   
   @doc("Release batch of Cloud Events. The server responds with an HTTP 200 status code if at least one event is successfully released. The response body will include the set of successfully released lockTokens, along with other failed lockTokens with their corresponding error information.")
-  @route("/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:release")
-  @post op ReleaseCloudEvents is Azure.Core.RpcOperation<{
-    @doc("content type")
-    @header("content-type")
-    contentType: "application/json; charset=utf-8";
-
-    @doc("Topic Name.")
-    @path
-    topicName: string;
-
-    @doc("Event Subscription Name.")
-    @path
-    eventSubscriptionName: string;
-
-    @doc("ReleaseOptions")
-    @body    
-    lockTokens : ReleaseOptions;
+  @action("release")
+  op releaseCloudEvents is StandardOperations.ResourceAction<
+    EventSubscription,
+    {
+      @doc("content type")
+      @header("content-type")
+      contentType: "application/json; charset=utf-8";
```

Comment:
Nitpick; specifying the charset for content-type application/json is a bit redundant since json is supposed to be in UTF-8. Does the service actually require us to include the charset? If so, we should open a bug...

Reply:
Tested without and worked, will remove